domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/TabShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,102 @@
+    @Override
+    public ValidationResult validate(DisplayGroupEntity thisDisplayGroup, List<DisplayGroupEntity> allDisplayGroups) {
+        ValidationResult validationResult = new ValidationResult();
+        List allTabDisplayGroups = getAllTabDisplayGroups(thisDisplayGroup, allDisplayGroups);
+
+        if (tabPreconditions(thisDisplayGroup)) {
+            ShowCondition showCondition;
+            try {
+                showCondition = showConditionParser.parseShowCondition(thisDisplayGroup.getShowCondition());
+            } catch (InvalidShowConditionException e) {
+                validationResult.addError(new DisplayGroupInvalidTabShowCondition(thisDisplayGroup));
+                return validationResult;
+            }
+
+            String showConditionField = showCondition.getField();
+
+            if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                validationResult.addError(new DisplayGroupInvalidTabShowCondition(showConditionField, thisDisplayGroup));
+            }
+        }
+
+        if (tabFieldsPreconditions(thisDisplayGroup)) {
+            for (DisplayGroupCaseFieldEntity caseField : thisDisplayGroup.getDisplayGroupCaseFields()) {
+                if (caseField.getShowCondition() != null) {
+                    ShowCondition showCondition;
+                    try {
+                        showCondition = showConditionParser.parseShowCondition(caseField.getShowCondition());
+                    } catch (InvalidShowConditionException e) {
+                        validationResult.addError(new DisplayGroupInvalidTabFieldShowCondition(caseField));
+                        return validationResult;
+                    }
+
+                    String showConditionField = showCondition.getField();
+
+                    if (!isInTabDisplayGroups(allTabDisplayGroups, showConditionField)) {
+                        validationResult.addError(new DisplayGroupInvalidTabFieldShowCondition(showConditionField, caseField));
+                    }
+                }
+            }
+        }
+        return validationResult;
+    }
+
+    private List getAllTabDisplayGroups(DisplayGroupEntity thisDisplayGroup, Collection<DisplayGroupEntity> allDisplayGroups) {
+        return allDisplayGroups
+            .stream()
+            .filter(dg -> dg.getType() == DisplayGroupType.TAB)
+            .filter(dg -> dg.getCaseType().getReference().equals(thisDisplayGroup.getCaseType().getReference()))
+            .collect(Collectors.toList());
+    }
+
+    private boolean isInTabDisplayGroups(List<DisplayGroupEntity> tabDisplayGroups, String showConditionField) {
+        return tabDisplayGroups.stream().filter(tdg -> tdg.hasField(showConditionField)).findFirst().isPresent();
+    }
+
+    public boolean tabPreconditions(DisplayGroupEntity displayGroup) {
+        List<Predicate<DisplayGroupEntity>> preconditions = new ArrayList<>();
+        preconditions.add(dg -> !StringUtils.isBlank(dg.getShowCondition()));
```

Comment:
perhaps we could introduce a method hasShowCondition to DisplayGroupEntity?

Reply:
good point